webpack/JobInvocationDetail/JobInvocationHostTable.js

```diff
@@ -0,0 +1,150 @@
+  JOB_INVOCATION_HOSTS,
+  STATUS_UPPERCASE,
+} from './JobInvocationConstants';
+
+const JobInvocationHostTable = ({ id, targeting }) => {
+  const columns = Columns();
+  const columnNamesKeys = Object.keys(columns);
+  const apiOptions = { key: JOB_INVOCATION_HOSTS, search: urlSearchQuery };
+  const {
+    search: urlSearchQuery = '',
+    page: urlPage,
+    per_page: urlPerPage,
+  } = useUrlParams();
+  const defaultParams = { search: urlSearchQuery };
+  if (urlPage) defaultParams.page = Number(urlPage);
+  if (urlPerPage) defaultParams.per_page = Number(urlPerPage);
+  const { response, status, setAPIOptions } = useAPI(
+    'get',
+    `/api/job_invocations/${id}/hosts`,
+    {
+      params: { ...defaultParams, key: JOB_INVOCATION_HOSTS },
+    }
+  );
+
+  const { setParamsAndAPI, params } = useSetParamsAndApiAndSearch({
+    defaultParams,
+    apiOptions,
+    setAPIOptions,
+  });
+
+  const combinedResponse = {
+    response: {
+      search: '',
+      can_create: false,
+      results: response?.results || [],
+      total: response?.total || 0,
+      per_page: response?.perPage,
+      page: response?.page,
+      subtotal: response?.subtotal || 0,
+      message: response?.message || 'error',
+    },
+    status,
+    setAPIOptions,
+  };
+
+  const { updateSearchQuery } = useBulkSelect({
+    initialSearchQuery: urlSearchQuery,
+  });
+
+  const controller = 'hosts';
+  const memoDefaultSearchProps = useMemo(
+    () => getControllerSearchProps(controller),
+    [controller]
+  );
+  memoDefaultSearchProps.autocomplete.url = foremanUrl(
+    `/${controller}/auto_complete_search`
+  );
+
+  const getEmptyState = () => {
+    if (status === STATUS_UPPERCASE.RESOLVED && !response?.results?.length) {
+      if (targeting?.targeting_type === 'dynamic_query') {
+        return {
+          action: {
+            title: 'View hosts',
+            url: `/new/hosts?search=${targeting?.search_query}`,
+          },
+          message: sprintf(
+            __(
+              'The dynamic query is still being processed. You can view the hosts targeted by the query.'
+            )
+          ),
+        };
+      }
+      return {
+        action: null,
+        message: __('No hosts found'),
+      };
+    }
+    return {
+      action: null,
+      message: null,
+    };
+  };
+  const { action, message } = getEmptyState();
+
+  return (
+    <TableIndexPage
+      apiUrl=""
+      apiOptions={apiOptions}
+      customSearchProps={memoDefaultSearchProps}
+      controller="hosts"
+      creatable={false}
+      replacementResponse={combinedResponse}
+      updateSearchQuery={updateSearchQuery}
+    >
+      <Table
+        ouiaId="job-invocation-hosts-table"
+        columns={columns}
+        params={params}
+        setParams={setParamsAndAPI}
+        itemCount={response?.subtotal}
+        results={response?.results}
+        url=""
+        refreshData={() => {}}
+        emptyAction={action}
+        emptyMessage={message}
```

Comment:
I'm not sure an action with a button would be best here, since then the user cant open it in a new tab. instead of an action can there be a link in the message?

Reply:
I can change it as you're proposing, but I'll have to use my own `EmptyState` in this file or make the `emptyMessage` accept an object instead of a string. Is it okay?

Reply:
Yeah I think it should use its own EmptyState